(#74)(ENGTASKS-3605) Update ReadMe and Vagrantfile in preperation for release of new box version with Hyper-V provider (#77)
chocolatey-community/chocolatey-test-environment: ReadMe.md

For more information on vagrant commands, see the [Vagrant Docs](http://docs.vagrantup.com/v2/cli/index.html)

## Using Hyper-V instead of VirtualBox

> [!CAUTION]

> > It is recommended to favour running the Chocolatey Test Environment under VirtualBox.

Reviewer: ```suggestion
> It is recommended to favor running the Chocolatey Test Environment under VirtualBox.

```

Reviewer: We should use US spelling.

> Hyper-V support is intended to lower the barrier to entry for those using Windows features which force the usage of Hyper-V,
> such as the Windows Subsystem for Linux (WSL 2), Windows Sandbox, Device Guard, and Credential Guard.
>
> Please be sure that you're aware of the [limitations](https://developer.hashicorp.com/vagrant/docs/providers/hyperv/limitations) of using the Vagrant Hyper-V provider.

> [!NOTE]
> The Hyper-V role requires Windows 10 or 11 Enterprise, Pro, or Education.
> It **cannot** be installed on Home editions of Windows.

>  1. Ensure setup above is complete, ignoring the need for VirtualBox
>  1. Enable the Hyper-V feature
>      * Open a PowerShell console as Administrator
>      * Run `Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V-All -All`
>      * Restart Windows
>  1. Follow the general [Preparing the Testing Environment](#preparing-the-testing-environment) instructions except swap the default `vagrant up` command for `vagrant up --provider=hyperv`
>      * The image downloaded the first time you run this command is specific to Hyper-V and so a large download will be required even if you've already downloaded the VirtualBox image

Reviewer: ```suggestion
 1. Ensure setup above is complete, ignoring the need for VirtualBox.
 1. Enable the Hyper-V feature:
     * Open a PowerShell console as Administrator.
     * Run `Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Hyper-V-All -All`.
     * Restart Windows.
 1. Follow the general instructions to [prepare the testing environment](#preparing-the-testing-environment), swapping the default `vagrant up` command with `vagrant up --provider=hyperv`.
     * The Hyper-V image, downloaded the first time you run command is very large (> 5GB) and specific to Hyper-V.
```

## Differences Between This and Package Verifier Service

> There are a couple of difference between the [verifier service](https://docs.chocolatey.org/en-us/community-repository/moderation/package-verifier) and this environment.

Reviewer: ```suggestion
There are a couple of difference between the [Package Verifier service](https://docs.chocolatey.org/en-us/community-repository/moderation/package-verifier) and this environment.

```

 * The verifier is run without the GUI - meaning it is run in a headless state. There is no box to interact with.

>  * The verifier only runs against Windows 2019 currently. This repo is adding more boxes as they become available.

Reviewer: ```suggestion
 * Package Verifier only runs against Windows Server 2019, currently.

```

>  * The verifier times out on waiting for a command after 12 minutes.

Reviewer: ```suggestion
 * Package Verifier times out on waiting for a command after 12 minutes.

```

>  * Synced folders are different - the verifier syncs the .chocolatey folder to gather the package information files.

Reviewer: ```suggestion
 * Synced folders are different - Package Verifier syncs the `.chocolatey` folder to gather the package information files.

```
